### PR TITLE
Correct default in external dependencies patterns description

### PR DESCRIPTION
**Pull request by tenstad**
Ref https://github.com/jfrog/terraform-provider-artifactory/pull/263#issuecomment-1011268218

### CODE REVIEW

File: docs/resources/artifactory_remote_helm_repository.md [hunk 1]
```diff
@@ -28,5 +28,6 @@ All generic repo arguments are supported, in addition to:
 * `helm_charts_base_url` - (Optional) - No documentation is available. Hopefully you know what this means
 * `external_dependencies_enabled` - (Optional) When set, external dependencies are rewritten.
 * `external_dependencies_patterns` - (Optional) An Allow List of Ant-style path expressions that specify where external
-  dependencies may be downloaded from. By default, this is set to ** which means that dependencies may be downloaded
-  from any external source.
+  dependencies may be downloaded from. By default, this is an empty list which means that no dependencies may be downloaded
+  from external sources. Note that the official documentation states the default is '**', which is correct when creating
+  repositories in the UI, but incorrect for the API.
```
Comment (chb0github) [inline docs/resources/artifactory_remote_helm_repository.md:33]: WOW!!!!!! I mean, I am shocked but not suprised.

**Comment by chb0github** [comment]
@alexhung - Here's the open question: We can set the default in TF to match the behavior of the UI `**`. What do you think?

**Comment by alexhung** [comment]
@chb0github We can do that too. Probably provide a better user experience, at the cost of us potentially need to keep in sync with any changes of behavior in the UI.

**Comment by tenstad** [comment]
How would you define the default value when TF schemas does not (AFAIK) support default values for list elemets? Some additional logic to make it `["**"]` if it is undefined?

The defaults (API/UI) seem to be equal for remote docker repositories as well. Probably the case for all types and not just helm.

**Comment by chb0github** [comment]
TF allows you to set a default. In addition, we can explicitly do it in code. Either way, It's easy

**Comment by chb0github** [comment]
@alexhung - a users expectation is one of consistency - the delta between the UI and the API (the API documentation is not nearly as well kept) would definitely be problematic. I mean, where do you think I got all the description information for the fields? I literally ripped it from the UI tooltip

**Comment by chb0github** [comment]
The difference isn't different from the official docs - I wouldn't consider the UI tooltip "official" - but the behavior observed when things are done through the UI is crucial and effectively becomes the standard.

**Comment by tenstad** [comment]
I read https://www.jfrog.com/confluence/display/JFROG/Kubernetes+Helm+Chart+Repositories when trying to get external dependencies working for my helm remote. For _Pattern Allow List_ it states _By default, this is set to ** which means that dependencies may be downloaded from any external source_. I guess it assumes creation through UI. Tried to verify with https://www.jfrog.com/confluence/display/JFROG/Repository+Configuration+JSON, but there (for remote) `externalDependenciesPatterns` _only applies to Docker repositories_.

**Review by chb0github — APPROVED** [review]

**Comment by tenstad** [comment]
Sadly, yeah :/ Here is an issue for it: https://github.com/hashicorp/terraform-plugin-sdk/issues/142

**Comment by alexhung** [comment]
@chb0github I'm not disagreeing with you at all :) I think we should be consistent, just need to acknowledge that there is a cost to keep this consistent with the UI.

**Comment by chb0github** [comment]
@tenstad - so you have to declare an empty block? I mean, at that point, just declare a value!

It can be done in code too: 
```go
if ok,val := d.get("external_dependencies_patterns"); !ok {
    payload.ExternalDependencyPatterns = []string{"**"}
}
```
Not so sexy, but it would work

**Comment by chb0github** [comment]
@alexhung - your call to merge this

**Comment by alexhung** [comment]
@tenstad I'm going to merge this since this accurately reflects the behavior of the provider currently. I'll create a JIRA ticket for the wish to match UI behavior.

**Pull request merged**